### PR TITLE
Set dataset ID attribute

### PR DESCRIPTION
**Pull request by jrleeman**
Supersedes #233

### CODE REVIEW

File: siphon/catalog.py [hunk 1]
```diff
@@ -454,6 +454,12 @@ def __init__(self, element_node, catalog_url=''):
 
         """
         self.name = element_node.attrib['name']
+
+        if 'ID' in element_node.attrib:
+            self.id = element_node.attrib['ID']
```
Comment (dopplershift) [inline siphon/catalog.py]: `self.id = element_node.attrib.get('ID', None)` would be shorter... 😁

Comment (dopplershift) [reply siphon/catalog.py]: Could technically do that to the `'urlPath'` block below if you're feeling motivated...

File: siphon/catalog.py [hunk 2]
```diff
@@ -454,8 +454,10 @@ def __init__(self, element_node, catalog_url=''):
 
         """
         self.name = element_node.attrib['name']
+
         if 'urlPath' in element_node.attrib:
             self.url_path = element_node.attrib['urlPath']
+            self.id = element_node.attrib.get('ID', None)
```
Comment (dopplershift) [inline siphon/catalog.py]: Did you really mean to move this line *into* the block? What I meant was that you could do, outside any if block
```python
self.id = element_node.attrib.get('ID', None)
self.url_path = element_node.attrib.get('urlPath', None)
```

File: siphon/tests/test_catalog.py [hunk 3]
```diff
@@ -268,9 +268,10 @@ def test_simple_service_within_compound():
     url = ('http://thredds-test.unidata.ucar.edu/thredds/catalog/noaaport/text/'
            'tropical/atlantic/hdob/catalog.xml')
     cat = TDSCatalog(url)
-    assert (cat.datasets[0].access_urls ==
-            {'HTTPServer': 'http://thredds-test.unidata.ucar.edu/thredds/fileServer/noaaport/'
-                           'text/tropical/atlantic/hdob/High_density_obs_20170824.txt'})
+    assert (cat.datasets[0].access_urls
+            == {'HTTPServer': 'http://thredds-test.unidata.ucar.edu/thredds/'
+                              'fileServer/noaaport/text/tropical/atlantic/hdob/''
```
Comment (dopplershift) [inline siphon/tests/test_catalog.py]: Oops, note the `''` at the end of the line.

**Comment by jrleeman** [comment]
Ok - I think everything is happy here now @dopplershift

**Pull request merged**